### ISSUE TITLE
WARNING:tensorflow:A checkpoint was restored (e.g. tf.train.Checkpoint.restore or tf.keras.Model.load_weights) but not all checkpointed values were used. See above for specific issues. Use expect_partial() on the load status object, e.g. tf.train.Checkpoint.restore(...).expect_partial(), to silence these warnings, or use assert_consumed() to make the check explicit. See https://www.tensorflow.org/guide/checkpoint#loading_mechanics for details

### ISSUE THREAD

**Issue by Deepthi-Jain** [tensorflow/tensorflow#41862]
![Capture](https://user-images.githubusercontent.com/62254038/88812450-a242f300-d1af-11ea-8d2f-e461a0e2c00e.JPG)

Hi All,

i'm trying to train train my own models with the tensorflow 2.2, by using predefined models used this link but i'm facing the above error. Can anyone please let me know anyone was able to run the pretrained models on tensorflow 2.2??

thanks in advance.

**Comment by ravikyram** [comment]
@Deepthi-Jain 

Please, fill [issue template.](https://github.com/tensorflow/tensorflow/issues/new/choose).

Please, let us know which pretrained model you are using and share related code .Thanks!

**Comment by sathyamsn** [comment]
Having similar issue with efficientdet_d1_coco17_tpu-32 pre-trained model.

**Comment by turowicz** [comment]
@ravikyram I'm having the same issue with TF 2.3.0 Object Detection and Efficientnet D1

**Comment by vamshi-rvk** [comment]
> @ravikyram I'm having the same issue with TF 2.3.0 Object Detection and Efficientnet D1
this will fix it
  fine_tune_checkpoint_type: "fine_tune" instead of fine_tune_checkpoint_type: "detection"

**Comment by ali1366** [comment]
I had the same problem. Changing  fine_tune_checkpoint_type: "classification"  to  fine_tune_checkpoint_type: "detection" in the config file, solved my problem. After solving the above problem, training job work but my model did not converge so I decreased the leraning_rate and momentum. After decreasing the learning_rate and momentum my model converged.
 NOTE: If you got a resource exhausted error (OOM error) decrease the batch_size in the config file.
The final config file is like below:
```
# SSD with Resnet 50 v1 FPN feature extractor, shared box predictor and focal
# loss (a.k.a Retinanet).
# See Lin et al, https://arxiv.org/abs/1708.02002
# Trained on COCO, initialized from Imagenet classification checkpoint
# Train on TPU-8
#
# Achieves 34.3 mAP on COCO17 Val

model {
  ssd {
    inplace_batchnorm_update: true
    freeze_batchnorm: true
    num_classes: 3
    box_coder {
      faster_rcnn_box_coder {
        y_scale: 10.0
        x_scale: 10.0
        height_scale: 5.0
        width_scale: 5.0
      }
    }
    matcher {
      argmax_matcher {
        matched_threshold: 0.5
        unmatched_threshold: 0.5
        ignore_thresholds: false
        negatives_lower_than_unmatched: true
        force_match_for_each_row: true
        use_matmul_gather: true
      }
    }
    similarity_calculator {
      iou_similarity {
      }
    }
    encode_background_as_zeros: true
    anchor_generator {
      multiscale_anchor_generator {
        min_level: 3
        max_level: 7
        anchor_scale: 4.0
        aspect_ratios: [1.0, 2.0, 0.5]
        scales_per_octave: 2
      }
    }
    image_resizer {
      fixed_shape_resizer {
        height: 640
        width: 640
      }
    }
    box_predictor {
      weight_shared_convolutional_box_predictor {
        depth: 256
        class_prediction_bias_init: -4.6
        conv_hyperparams {
          activation: RELU_6,
          regularizer {
            l2_regularizer {
              weight: 0.0004
            }
          }
          initializer {
            random_normal_initializer {
              stddev: 0.01
              mean: 0.0
            }
          }
          batch_norm {
            scale: true,
            decay: 0.997,
            epsilon: 0.001,
          }
        }
        num_layers_before_predictor: 4
        kernel_size: 3
      }
    }
    feature_extractor {
      type: 'ssd_resnet50_v1_fpn_keras'
      fpn {
        min_level: 3
        max_level: 7
      }
      min_depth: 16
      depth_multiplier: 1.0
      conv_hyperparams {
        activation: RELU_6,
        regularizer {
          l2_regularizer {
            weight: 0.0004
          }
        }
        initializer {
          truncated_normal_initializer {
            stddev: 0.03
            mean: 0.0
          }
        }
        batch_norm {
          scale: true,
          decay: 0.997,
          epsilon: 0.001,
        }
      }
      override_base_feature_extractor_hyperparams: true
    }
    loss {
      classification_loss {
        weighted_sigmoid_focal {
          alpha: 0.25
          gamma: 2.0
        }
      }
      localization_loss {
        weighted_smooth_l1 {
        }
      }
      classification_weight: 1.0
      localization_weight: 1.0
    }
    normalize_loss_by_num_matches: true
    normalize_loc_loss_by_codesize: true
    post_processing {
      batch_non_max_suppression {
        score_threshold: 1e-8
        iou_threshold: 0.6
        max_detections_per_class: 100
        max_total_detections: 100
      }
      score_converter: SIGMOID
    }
  }
}

train_config: {
  fine_tune_checkpoint_version: V2
  fine_tune_checkpoint: "./test_data/ssd_resnet50/checkpoint/ckpt-0"
  fine_tune_checkpoint_type: "detection"
  batch_size: 10
  sync_replicas: true
  startup_delay_steps: 0
  replicas_to_aggregate: 5
  use_bfloat16: true
  num_steps: 100000
  data_augmentation_options {
    random_horizontal_flip {
    }
  }
  data_augmentation_options {
    random_crop_image {
      min_object_covered: 0.0
      min_aspect_ratio: 0.4
      max_aspect_ratio: 3.0
      min_area: 0.5
      max_area: 1.0
      overlap_thresh: 0.0
    }
  }
  optimizer {
    momentum_optimizer: {
      learning_rate: {
        cosine_decay_learning_rate {
          learning_rate_base: .0005
          total_steps: 100000
          warmup_learning_rate: .0001
          warmup_steps: 1000
        }
      }
      momentum_optimizer_value: 0.5
    }
    use_moving_average: false
  }
  max_number_of_boxes: 100
  unpad_groundtruth_tensors: false
}

train_input_reader: {
  label_map_path: "./test_images/label_map.pbtxt"
  tf_record_input_reader {
    input_path: "./test_images/train.record-00000-of-00001"
  }
}

eval_config: {
  metrics_set: "coco_detection_metrics"
  use_moving_averages: false
}

eval_input_reader: {
  label_map_path: "./test_images/label_map.pbtxt"
  shuffle: false
  num_epochs: 1
  tf_record_input_reader {
    input_path:  "./test_images/val.record-00000-of-00001"
  }
}

**Comment by sidharth1805** [comment]
> I had the same problem. Changing fine_tune_checkpoint_type: "classification" to fine_tune_checkpoint_type: "detection" in the config file, solved my problem. After solving the above problem, training job work but my model did not converge so I decreased the leraning_rate and momentum. After decreasing the learning_rate and momentum my model converged.
> NOTE: If you got a resource exhausted error (OOM error) decrease the batch_size in the config file.
> The final config file is like below:
> 
> ```
> # SSD with Resnet 50 v1 FPN feature extractor, shared box predictor and focal
> # loss (a.k.a Retinanet).
> # See Lin et al, https://arxiv.org/abs/1708.02002
> # Trained on COCO, initialized from Imagenet classification checkpoint
> # Train on TPU-8
> #
> # Achieves 34.3 mAP on COCO17 Val
> 
> model {
>   ssd {
>     inplace_batchnorm_update: true
>     freeze_batchnorm: true
>     num_classes: 3
>     box_coder {
>       faster_rcnn_box_coder {
>         y_scale: 10.0
>         x_scale: 10.0
>         height_scale: 5.0
>         width_scale: 5.0
>       }
>     }
>     matcher {
>       argmax_matcher {
>         matched_threshold: 0.5
>         unmatched_threshold: 0.5
>         ignore_thresholds: false
>         negatives_lower_than_unmatched: true
>         force_match_for_each_row: true
>         use_matmul_gather: true
>       }
>     }
>     similarity_calculator {
>       iou_similarity {
>       }
>     }
>     encode_background_as_zeros: true
>     anchor_generator {
>       multiscale_anchor_generator {
>         min_level: 3
>         max_level: 7
>         anchor_scale: 4.0
>         aspect_ratios: [1.0, 2.0, 0.5]
>         scales_per_octave: 2
>       }
>     }
>     image_resizer {
>       fixed_shape_resizer {
>         height: 640
>         width: 640
>       }
>     }
>     box_predictor {
>       weight_shared_convolutional_box_predictor {
>         depth: 256
>         class_prediction_bias_init: -4.6
>         conv_hyperparams {
>           activation: RELU_6,
>           regularizer {
>             l2_regularizer {
>               weight: 0.0004
>             }
>           }
>           initializer {
>             random_normal_initializer {
>               stddev: 0.01
>               mean: 0.0
>             }
>           }
>           batch_norm {
>             scale: true,
>             decay: 0.997,
>             epsilon: 0.001,
>           }
>         }
>         num_layers_before_predictor: 4
>         kernel_size: 3
>       }
>     }
>     feature_extractor {
>       type: 'ssd_resnet50_v1_fpn_keras'
>       fpn {
>         min_level: 3
>         max_level: 7
>       }
>       min_depth: 16
>       depth_multiplier: 1.0
>       conv_hyperparams {
>         activation: RELU_6,
>         regularizer {
>           l2_regularizer {
>             weight: 0.0004
>           }
>         }
>         initializer {
>           truncated_normal_initializer {
>             stddev: 0.03
>             mean: 0.0
>           }
>         }
>         batch_norm {
>           scale: true,
>           decay: 0.997,
>           epsilon: 0.001,
>         }
>       }
>       override_base_feature_extractor_hyperparams: true
>     }
>     loss {
>       classification_loss {
>         weighted_sigmoid_focal {
>           alpha: 0.25
>           gamma: 2.0
>         }
>       }
>       localization_loss {
>         weighted_smooth_l1 {
>         }
>       }
>       classification_weight: 1.0
>       localization_weight: 1.0
>     }
>     normalize_loss_by_num_matches: true
>     normalize_loc_loss_by_codesize: true
>     post_processing {
>       batch_non_max_suppression {
>         score_threshold: 1e-8
>         iou_threshold: 0.6
>         max_detections_per_class: 100
>         max_total_detections: 100
>       }
>       score_converter: SIGMOID
>     }
>   }
> }
> 
> train_config: {
>   fine_tune_checkpoint_version: V2
>   fine_tune_checkpoint: "./test_data/ssd_resnet50/checkpoint/ckpt-0"
>   fine_tune_checkpoint_type: "detection"
>   batch_size: 10
>   sync_replicas: true
>   startup_delay_steps: 0
>   replicas_to_aggregate: 5
>   use_bfloat16: true
>   num_steps: 100000
>   data_augmentation_options {
>     random_horizontal_flip {
>     }
>   }
>   data_augmentation_options {
>     random_crop_image {
>       min_object_covered: 0.0
>       min_aspect_ratio: 0.4
>       max_aspect_ratio: 3.0
>       min_area: 0.5
>       max_area: 1.0
>       overlap_thresh: 0.0
>     }
>   }
>   optimizer {
>     momentum_optimizer: {
>       learning_rate: {
>         cosine_decay_learning_rate {
>           learning_rate_base: .0005
>           total_steps: 100000
>           warmup_learning_rate: .0001
>           warmup_steps: 1000
>         }
>       }
>       momentum_optimizer_value: 0.5
>     }
>     use_moving_average: false
>   }
>   max_number_of_boxes: 100
>   unpad_groundtruth_tensors: false
> }
> 
> train_input_reader: {
>   label_map_path: "./test_images/label_map.pbtxt"
>   tf_record_input_reader {
>     input_path: "./test_images/train.record-00000-of-00001"
>   }
> }
> 
> eval_config: {
>   metrics_set: "coco_detection_metrics"
>   use_moving_averages: false
> }
> 
> eval_input_reader: {
>   label_map_path: "./test_images/label_map.pbtxt"
>   shuffle: false
>   num_epochs: 1
>   tf_record_input_reader {
>     input_path:  "./test_images/val.record-00000-of-00001"
>   }
> }
> ```

Still having the same issue on faster r-cnn resnet 101

**Comment by aPeter1** [comment]
Still having the same issue on mobilnet as well
